File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/http/controllers/trainer/model/request/AddTraineeDataRequest.kt
package pt.isel.leic.ptgest.http.controllers.trainer.model.request

import pt.isel.leic.ptgest.domain.common.Gender
import pt.isel.leic.ptgest.domain.common.SkinFold
import pt.isel.leic.ptgest.domain.common.model.BodyCircumferences
import java.util.UUID

data class AddTraineeDataRequest(
    val traineeId: UUID,
    val gender: Gender,
    val weight: Double,
    val height: Double,
    val bodyCircumferences: BodyCircumferences,
    val bodyFatPercentage: Double?,
    val skinFold: Map<SkinFold, Double>?
)

File: code/js/src/views/user/trainerViews/models/traineeData/AddTraineeDataRequest.ts
import BodyCircumferences from "./BodyCircumferences.ts"

class AddTraineeDataRequest {
    traineeId: string
    gender: string
    weight: number
    height: number
    bodyCircumferences: BodyCircumferences
    bodyFatPercentage: number | null
    skinFold: Map<string, number> | null

    constructor(traineeId: string, gender: string, weight: number, height: number, bodyCircumferences: BodyCircumferences, bodyFatPercentage: number | null, skinFold: Map<string, number> | null) {
        this.traineeId = traineeId
        this.gender = gender
        this.weight = weight
        this.height = height
        this.bodyCircumferences = bodyCircumferences
        this.bodyFatPercentage = bodyFatPercentage
        this.skinFold = skinFold
    }
}
export default AddTraineeDataRequest
